server/src/main/resources/rules/rules.js

```diff
@@ -477,19 +477,23 @@ function get_mock_ent_for_pool(pool, consumer) {
 
 function get_pool_priority(pool, consumer) {
     log.debug("Calculating pool priority for pool " + pool.id + "...");
-    // start with a default
-    var priority = 100;
-
-    // list of syspurpose attributes and their corresponding weights.
+    // start with a default large enough to make sure that if all the highest syspurpose mismatch rules get
+    // applied (5600 * -0.5 +...+ 350 * -0.5 = -2800 -1400 -700 -350 -175 = -5425), the total score will not go below zero.
+    var priority = 5450;
+    log.debug("Starting with default initial score of "+priority);
+
+    // List of syspurpose attributes and their corresponding weights: the relationship between them,
+    // and the ones of less important attributes, is such that each one of them
+    // is larger than all the other attributes below it combined
+    // (e.g. usage: 350 > requires_host+virt_only+sockets+ram+cores+vcpu: 330).
     var attrs = {
-        'products': 20,
-        'roles': 10,
-        'addons': 5,
-        'support_level': 3,
-        'usage': 1
+        'products': 5600,
+        'roles': 2800,
+        'addons': 1400,
+        'support_level': 700,
+        'usage': 350
```

Comment:
Out of curiosity, where did these values come from?

Reply:
They were the initial values considered during the design, but for unknown reasons they were not used: https://docs.google.com/document/d/1q-gMSby30mDsTu0Yjlb3yV3Tt_x9KW_yZSY39lNsK0Q/edit#

But they are not entirely random, the relationship between the previous and the next weight is important, as the comment above them explains. If you add up all the max scores of all the attributes that are less important than usage(requires_host,virt_only,ram,cores,sockets,vcpu), they make up 330 points (150+100+20+20+20+20), so usage cannot be less than that sum. Same for support_level (330 + 350 < 700), and every one after that.

The idea is that each attribute is at least as important as all the attributes below it, combined.